### PR TITLE
Fix BUILD_CLONE_SUBMODULES CMake config option

### PR DESCRIPTION
**Pull request by aggarw13**
With submodule configuration changes made in #75, the `BUILD_CLONE_SUBMODULES` CMake config became disfynctional (as it uses the `git submodule update --recursive` command).

This PR updates the CMake logic to use `git submodule update --checkout --init` command to fix the CMake config option.
Also, this PR changes the default value of `BUILD_CLONE_SUBMODULES` to be `OFF` to avoid _always_ auto-cloning the **CMock** submodule.

### CODE REVIEW

**Review by yourslab — APPROVED** [review]

File: test/unit-test/cmock_build.cmake [hunk 1]
```diff
@@ -3,7 +3,7 @@ macro( clone_cmock )
         find_package( Git REQUIRED )
         message( "Cloning submodule CMock." )
         execute_process( COMMAND rm -rf ${CMOCK_DIR}
-                         COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive ${CMOCK_DIR}
+                         COMMAND ${GIT_EXECUTABLE} submodule update checkout --init --recursive ${CMOCK_DIR}
```
Comment (sarenameas) [inline test/unit-test/cmock_build.cmake:6]: Does this need to be --checkout?

Comment (sarenameas) [reply test/unit-test/cmock_build.cmake:6]: If we take out the clone you added to the ci.yml, we could test this?

Comment (aggarw13) [reply test/unit-test/cmock_build.cmake:6]: >Does this need to be --checkout?

Yes, that is how the `update = none` option is overriden

Comment (aggarw13) [reply test/unit-test/cmock_build.cmake:6]: > If we take out the clone you added to the ci.yml, we could test this?

Sure

**Review by sarenameas — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**